pkg/distributor/distributor_test.go

```diff
@@ -8684,12 +8789,13 @@ func checkRequestState(t *testing.T, ctx context.Context, acquiredPermit bool) {
 
 // mockReactiveLimiter is a mock implementation of ReactiveLimiter for testing
 type mockReactiveLimiter struct {
-	canAcquire bool
-	permit     *mockPermit
+	canAcquire    bool
+	failOnAcquire bool
+	permit        *mockPermit
 }
 
 func (m *mockReactiveLimiter) AcquirePermit(_ context.Context) (adaptivelimiter.Permit, error) {
-	if !m.canAcquire {
+	if m.failOnAcquire {
```

Comment:
nit: should this check both flags? i think in the current code we never call `AcquirePermit` if `canAcquire` is `false`, but having  `AcquirePermit` return a permit when `canAcquire` is `false`seems like a subtle thing that could cause issues down the line.